src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceResourceProvider.cs

```diff
@@ -34,8 +34,16 @@ public HttpSourceResourceProvider()
             Debug.Assert(source.PackageSource.IsHttp, "HTTP source requested for a non-http source.");
 
             HttpSourceResource curResource = null;
+            IThrottle throttle = NullThrottle.Instance;
 
-            var throttle = Throttle ?? NullThrottle.Instance;
+            if (Throttle != null)
+            {
+                throttle = Throttle;
```

Comment:
our docs should clearly says that `--disable-parallel` takes precedent over `MaxHttpRequests`